kustomize/commands/edit/set/setannotation.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package set
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/kustomize/api/konfig"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kustomize/v4/commands/internal/kustfile"
+	"sigs.k8s.io/kustomize/kustomize/v4/commands/internal/util"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+type setAnnotationOptions struct {
+	metadata     map[string]string
+	mapValidator func(map[string]string) error
+}
+
+// first part for prefix segment (DNS1123Label) of an annotation followed by a slash, second part for name segment of an annotation
+var IsValidKey = regexp.MustCompile(`^([a-z0-9]([-a-z0-9]*[a-z0-9])?\/)?[a-zA-Z0-9]([-a-zA-Z0-9_.]*[a-zA-Z0-9])?$`).MatchString
```

Comment:
if we're going to do this validation, can we also add to the regexp that the length of the prefix must be {0,253} chars and length of the name must be {0,63} chars? This isn't a big deal and we can do without it but it's simple enough to add I think

Reply:
Please check the new regex carefully :)